Snake/BusinessLayer/maze.cs

```diff
@@ -40,20 +52,87 @@ public maze(int w,int h)
                 {
                     for (int j = 0; j < width; j++)
                     {
-                        maze[i, j] = border;
-                    } 
+                        Maze[i, j] = BORDER;
+                    }
                 }
 
                 //first and last colum of other rows should be marked as border
                 else
                 {
-                    maze[i, 0] = border;
-                    maze[i, width - 1] = border;
+                    Maze[i, 0] = BORDER;
+                    Maze[i, width - 1] = BORDER;
                 }
             }
 
-            return maze;
         }
 
+
+
+        public void Generate_All_Obstacles()
```

Comment:
I would avoid underscores in method and variable names. GenerateAllObstacles is fine to read.